File: explorer/src/main/java/com/duongame/helper/ExtSdCardHelper.java
package com.duongame.helper;

import android.os.Environment;

import java.io.File;
import java.io.InputStream;
import java.util.HashSet;
import java.util.Locale;

import static android.os.Environment.getExternalStorageDirectory;
import static com.duongame.file.FileHelper.BLOCK_SIZE;

/**
 * Created by namjungsoo on 2017-01-22.
 */

public class ExtSdCardHelper {
    public static String getExternalSdCardPath() {
        File root = Environment.getExternalStorageDirectory();
        if (root == null)
            return null;

        File parent = root.getParentFile();
        if (parent == null)
            return null;

        File storage = parent.getParentFile();
        if (storage == null)
            return null;

        File[] files = storage.listFiles();
        if(files == null)
            return null;

        for (File file : files) {
            String path = file.getName();
            if (path.equals("emulated"))
                continue;
            if (path.equals("self"))
                continue;
            return file.getAbsolutePath();
        }
        return null;
    }
}

File: Explorer/src/main/java/com/duongame/helper/ExtSdCardHelper.kt
package com.duongame.helper

import android.os.Environment

/**
 * Created by namjungsoo on 2017-01-22.
 */
object ExtSdCardHelper {
    @JvmStatic
    val externalSdCardPath: String?
        get() {
            val root = Environment.getExternalStorageDirectory() ?: return null
            val parent = root.parentFile ?: return null
            val storage = parent.parentFile ?: return null
            val files = storage.listFiles() ?: return null
            for (file in files) {
                val path = file.name
                if (path == "emulated") continue
                if (path == "self") continue
                return file.absolutePath
            }
            return null
        }
}